### PR TITLE
Bugfix: colorUpdate on embeddings plot, classification in modal

### PR DESCRIPTION
**Pull request by lanzhenw**
## What changes are proposed in this pull request?

1.  Embeddings plot use consistent color pool and special color setting based on label/attribute value.

Screenshot 2023-05-09 at 2 55 11 PM
Screenshot 2023-05-09 at 2 56 18 PM

2. fix bugs: classification bubble colors on sample modal view; tooltip border in dark mode; 

## How is this patch tested? If it is not, please explain why.

(Details)

### Is this a user-facing change that should be mentioned in the release notes?

-   [ ] No. You can skip the rest of this section.
-   [ ] Yes. Give a description of this change to be included in the release
        notes for FiftyOne users.

(Details in 1-2 sentences. You can just refer to another PR with a description
if this PR is part of a larger change.)

### What areas of FiftyOne does this PR affect?

-   [x] App: FiftyOne application changes
-   [ ] Build: Build and test infrastructure changes
-   [ ] Core: Core `fiftyone` Python library changes
-   [ ] Documentation: FiftyOne documentation changes
-   [ ] Other

### CODE REVIEW

File: app/packages/embeddings/src/tracesToData.tsx [hunk 1]
```diff
@@ -39,7 +42,7 @@ export function tracesToData(
           autocolorscale: !isContinuous, // isCategorical || isUncolored,
           colorscale: mappedColorscale,
           color: isCategorical
-            ? color.toCSSRGBString()
+            ? color?.toCSSRGBString()
```
Comment (imanjra) [inline app/packages/embeddings/src/tracesToData.tsx:45]: Is possibly of color being undefined expected/intended? Should there be fallback?

File: app/packages/embeddings/src/tracesToData.tsx [hunk 2]
```diff
@@ -70,3 +73,48 @@ export function tracesToData(
...
 }
+
+const getLabelColor = (key: string, setting: CustomizeColor) => {
+  if (setting && setting.labelColors) {
+    const color = setting.labelColors.find((x) => x.name === key)?.color;
+    return getConvertedColor(color);
+  }
+};
+
+const getConvertedColor = (color: string | number[]) => {
```
Comment (benjaminpkane) [inline app/packages/embeddings/src/tracesToData.tsx:84]: I think the return value here should be explicitly typed, or the conditionals cleaned up. It currently says it strictly returns `Color`

File: app/packages/embeddings/src/tracesToData.tsx [hunk 3]
```diff
@@ -70,3 +73,48 @@ export function tracesToData(
...
+      );
+    } else {
+      const idx = cssColorNames.name.map((x) => x.toLowerCase()).indexOf(color);
+      if (idx > -1) {
+        const c = hexToRgb(cssColorNames.hex[idx]);
+        return Color.fromCSSRGBValues(c.r, c.g, c.b);
+      }
+    }
+  } else {
+    if (color) {
```
Comment (benjaminpkane) [inline app/packages/embeddings/src/tracesToData.tsx:104]:
```suggestion
  } else if (Array.isArray(color) {
```

File: app/packages/embeddings/src/EmbeddingsPlot.tsx [hunk 4]
```diff
@@ -19,8 +17,14 @@ export function EmbeddingsPlot({
   plotSelection,
 }) {
   const theme = useTheme();
-  const getColor = useRecoilValue(fos.colorMapRGB(false));
-  const datasetName = useRecoilValue(fos.datasetName);
+  const getColor = useRecoilValue(fos.colorMap(false));
+  const customizedColorSettings = useRecoilValue(
+    fos.sessionColorScheme
+  ).customizedColorSettings;
+  const setting = customizedColorSettings.find((setting) =>
```
Comment (benjaminpkane) [inline app/packages/embeddings/src/EmbeddingsPlot.tsx:24]: Could be good to add use `useMemo`

File: app/packages/embeddings/src/tracesToData.tsx [hunk 5]
```diff
@@ -70,3 +73,48 @@ export function tracesToData(
       };
     });
 }
+
+const getLabelColor = (key: string, setting: CustomizeColor) => {
```
Comment (benjaminpkane) [inline app/packages/embeddings/src/tracesToData.tsx:77]: `getLabelColor` and `getConvertedColor` types and implementation seem to differ. Could lead to bugs

File: app/packages/embeddings/src/Color.tsx [hunk 6]
```diff
@@ -15,3 +15,306 @@ export class Color {
     return `rgb(${this.r * 255}, ${this.g * 255}, ${this.b * 255})`;
   }
 }
+
+export const cssColorNames = {
```
Comment (benjaminpkane) [inline app/packages/embeddings/src/Color.tsx:19]: Why is this needed? Could use a comment

**Review by benjaminpkane — CHANGES_REQUESTED** [review]
Left a few cosmetic comments/suggestions. But I think the important things is testing if the `Regression` label type is working. It is similar to `Classification` I think it should be tested too

**Comment by lanzhenw** [comment]
Regression Example: 
```
import random
import numpy as np

import fiftyone as fo
import fiftyone.zoo as foz
from fiftyone import ViewField as F

dataset = foz.load_zoo_dataset("quickstart").select_fields().clone()

# Populate some fake regression + weather data
for idx, sample in enumerate(dataset, 1):
    ytrue = random.random() * idx
    ypred = ytrue + np.random.randn() * np.sqrt(ytrue)
    confidence = random.random()
    sample["ground_truth"] = fo.Regression(value=ytrue)
    sample["predictions"] = fo.Regression(value=ypred, confidence=confidence)
    sample["weather"] = random.choice(["sunny", "cloudy", "rainy"])
    sample.save()

print(dataset)

results = dataset.evaluate_regressions("predictions", gt_field="ground_truth", eval_key="eval")

# Print some standard regression evaluation metrics
results.print_metrics()

# Plot a scatterplot of the results colored by `weather` and scaled by `confidence`
plot = results.plot_results(labels="weather", sizes="predictions.confidence")
plot.show()

session = fo.launch_app(dataset)

# Show the samples with the smallest regression error
session.view = dataset.sort_by("eval")

# Show the samples with the largest regression error
session.view = dataset.sort_by("eval", reverse=True)
```
Screenshot 2023-05-10 at 11 12 22 AM
Screenshot 2023-05-10 at 11 18 23 AM

File: app/packages/embeddings/src/tracesToData.tsx (hunk 1 above)
Comment (lanzhenw) [reply app/packages/embeddings/src/tracesToData.tsx:45]: I will add a fallback just in case.

File: app/packages/looker/src/overlays/classifications.ts [hunk 7]
```diff
@@ -49,17 +49,57 @@ export class ClassificationsOverlay<
...
-    return getColor(
-      state.options.coloring.pool,
-      state.options.coloring.seed,
-      key
-    );
+    const { coloring, customizeColorSetting } = state.options;
+    const f = field.startsWith("frames.")
+      ? field.slice("frames.".length)
+      : field;
+    switch (coloring.by) {
```
[the remark this answers is not in the document]
Comment (lanzhenw) [reply app/packages/looker/src/overlays/classifications.ts:63]: I just replaced with if.

File: app/packages/embeddings/src/Color.tsx (hunk 6 above)
Comment (lanzhenw) [reply app/packages/embeddings/src/Color.tsx:20]: Color can be valid css names now, so added this constant to convert the name to hex value.

**Review by benjaminpkane — APPROVED** [review]
LGTM! Did a little linting
